### DISCUSSION TITLE
Make php8.1 minimum requirement

### DISCUSSION THREAD

**Discussion by sreichel** [OpenMage/magento-lts#4317] [Polls]
Long story short ...

- PHP8.0 is end-of-life since Nov 2022 (no security support since Nov 2023)
- more and more tools rely on php8.1+
- Magento2 requires php8.1 since 2.4.6 (Feb 2023)
- yeah, skip php 8.0

Just as we have abandoned support for PHP7.3, I would like to ....

- https://github.com/OpenMage/magento-lts/discussions/2402
- https://github.com/OpenMage/magento-lts/discussions/2253
- https://github.com/OpenMage/magento-lts/discussions/3089

**Comment by sreichel** [comment]
Mhh ... not what i have expected. :grey_question: 

What are the resons to still want php7-support?

[Please] Letz moderize OM and dont rely on age old limitations.

**Comment by kiatng** [comment]
Is this for v20? I'm for supporting PHP7. No issue for v21 for min PHP8+.

**Comment by sreichel** [comment]
Why to support php7?

We dropped support for <7.4, why not do drop <8.1 - at the right time?

IMHO ... the right time is now.

I am a fan of backward-compatibilty, but changing the requirents to move forward isanother thing. Maybe I'd not have suggested this last year, but php7.4 and als php8.0 is EEOL for a long time now.

We are limitating ourself!

If you want to write an extension, you cant use php8 features as long you want to support latest OM-version, You also cant rely on other modern tools, that mostly require php8.1. THIS HURTS a lot.

I'd not suggest it, but you can still ignorephp-requirements during composer install.

**Comment by kiatng** [comment]
Would dropping support of PHP7 in v20 compliance with [RFC Release Schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)?

**Reply by sreichel** [reply]
imho

Release schedule is not bound to any requirements. 

(btw... we cannot use 21 as next major version ... skip and bump to v22?)

**Comment by JesperHerrloff** [comment]
Better to make 8.3 as Minimum requirement as 8.1 is End of Life later this year. 

I migrated a client to OpenMage 20 from Php 5.6 Magento 1.9.1 Last Weekend. 

I can probably help maintaining a bit of code as I am pretty familiar with the  code base with all core hacks done to make the plattform work for a car spare parts dealer :) 

I have not so much experience in working with other people on git so it would be a new experience.

**Reply by addison74** [reply]
Even if a PHP version is EOL, there are particular situations when the operating system development teams offer support and security for a certain period of time. In Debian Bullseye 11 the version shipped is 7.4, in Bookworm 12 the version is 8.2. The former has an LTS until 2026. 

I'm not saying we should wait until 2026, but I still believe that changes that create BC's should be notified in advance publicly and a time should be set when they are merged into the main branch. This time period can be mutually agreed by upon and represents a warning for developers to update their code. Obviously, a change must be tested for a while, for implementing this tasks we should have that PR in draft or something like not being merged before a date.

**Reply by addison74** [reply]
BTW with the help of AI or with inspiration from other derivative projects, the Prototype library can also be removed, similarly at a certain date when the change will be effective. It can be kept in a directory for a while and it is up to the developer to delete it, but OpenMage should no longer depend on these old libraries that are no longer maintained.

**Reply by addison74** [reply]
In this survey I voted for yes. My opinion that at some date this year we should upgrade to version 8.2 as the minimum in OpenMage. Debian Trexie will come with PHP 8.4, I running intensive tests with OpenMage but not in production. So far there are no errors and those I found I reported already and most of them are fixed.

**Reply by addison74** [reply]
I wouldn't worry too much about the PHP version, I would rather focus on ZF-1 Future on which we base our project too.

**Reply by JesperHerrloff** [reply]
> Even if a PHP version is EOL, there are particular situations when the operating system development teams offer support and security for a certain period of time. In Debian Bullseye 11 the version shipped is 7.4, in Bookworm 12 the version is 8.2. The former has an LTS until 2026.
> 
> I'm not saying we should wait until 2026, but I still believe that changes that create BC's should be notified in advance publicly and a time should be set when they are merged into the main branch. This time period can be mutually agreed by upon and represents a warning for developers to update their code. Obviously, a change must be tested for a while, for implementing this tasks we should have that PR in draft or something like not being merged before a date.

As a server admin you have to take some actions to keep software up to date. 

A link on howto update to newer php in Debian Bullseye 11. There is often a way to solve most software updates. 
https://www.linuxtuto.com/how-to-install-php-8-1-on-debian-11/

**Reply by JesperHerrloff** [reply]
> I wouldn't worry too much about the PHP version, I would rather focus on ZF-1 Future on which we base our project too.

I completely agree with that. Will have to look into uses of the Zend Framework to see if its a viable solution to make some adapters to adapt to another framework.

**Reply by sreichel** [reply]
Zend usage .... https://github.com/openmage-strict/openmage-future/issues/6

**Comment by sreichel** [comment]
Maybe add something like 

- see https://github.com/OpenMage/Web_Notifications/pull/7 ... ?

**Reply by addison74** [reply]
Web Notifications is a powerful tool to announce the users about these major changes. I will approve it right away.
